### PR TITLE
[ETFE-4005] Drill-down on EVN-11: Report of receipt submitted

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: app/viewmodels/helpers/events/MovementEventHelper.scala [hunk 1]
```diff
@@ -459,4 +471,235 @@ class MovementEventHelper @Inject()(
...
+            Some(summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorDetails.dateOfArrival", reportOfReceipt.dateOfArrival.formatDateForUIOutput())),
+            Some(summaryListRowBuilder(statusKey, messages(s"movementHistoryEvent.${event.eventType}.rorDetails.status.value.${reportOfReceipt.acceptMovement}"))),
+            Some(summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorDetails.moreInformation", reportOfReceipt.otherInformation.getOrElse(s"movementHistoryEvent.${event.eventType}.rorDetails.moreInformation.notProvided"))),
+          )
+        )
+      case None => Html("")
+    }
+  }
+
+  def rorConsigneeCard(event: MovementHistoryEvent)(implicit movement: GetMovementResponse, messages: Messages): Html = {
```
Comment (MJCallahanPage) [inline app/viewmodels/helpers/events/MovementEventHelper.scala:505]: Couldn't the existing `Consignee` card be updated for this? I'd have thought this content would have been consistent everwhere?

File: app/viewmodels/helpers/events/MovementEventHelper.scala [hunk 2]
```diff
@@ -459,4 +471,235 @@ class MovementEventHelper @Inject()(
...
+          consigneeTrader.address.map(address => summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorConsignee.address", renderAddress(address))),
+          consigneeTrader.eoriNumber.map(summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorConsignee.eoriNumber", _))
+        )
+      )
+    }
+
+    optContent.getOrElse(Html(""))
+  }
+
+  def rorDestinationCard(event: MovementHistoryEvent)(implicit movement: GetMovementResponse, messages: Messages): Html = {
```
Comment (MJCallahanPage) [inline app/viewmodels/helpers/events/MovementEventHelper.scala:544]: Couldn't the existing `Place of Destination` card be updated for this? I'd have thought this content would have been consistent everywhere?

File: app/viewmodels/helpers/events/MovementEventHelper.scala [hunk 3]
```diff
@@ -459,4 +471,235 @@ class MovementEventHelper @Inject()(
...
+          deliveryPlaceTrader.vatNumber.map(summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorDeliveryPlaceTrader.vatNumber", _)),
+          deliveryPlaceTrader.address.map(address => summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorDeliveryPlaceTrader.address", renderAddress(address))),
+        )
+      )
+    }
+
+    optContent.getOrElse(Html(""))
+  }
+
+  def rorExportCard(event: MovementHistoryEvent)(implicit movement: GetMovementResponse, messages: Messages): Html = {
```
Comment (MJCallahanPage) [inline app/viewmodels/helpers/events/MovementEventHelper.scala:574]: Couldn't the existing `Export` information card be updated for this? I'd have thought this content would have been consistent everywhere?

File: app/viewmodels/helpers/events/MovementEventHelper.scala [hunk 4]
```diff
@@ -459,4 +471,235 @@ class MovementEventHelper @Inject()(
...
+          // TODO: check if destinationOffice is correct
+          Some(summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorExport.customsOfficeCode", reportOfReceipt.destinationOffice))
+        )
+      )
+    }
+
+    optContent.getOrElse(Html(""))
+  }
+
+  def rorItemsCard(event: MovementHistoryEvent, ie818ItemModelWithCnCodeInformation: Seq[IE818ItemModelWithCnCodeInformation])
```
Comment (MJCallahanPage) [inline app/viewmodels/helpers/events/MovementEventHelper.scala:594]: Couldn't the existing `Items` information card be updated for this? I'd have thought this content would have been consistent everywhere?

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]

File: app/viewmodels/helpers/events/MovementEventHelper.scala [hunk 5]
```diff
@@ -459,4 +471,235 @@ class MovementEventHelper @Inject()(
...
+    }
+
+    optContent.getOrElse(Html(""))
+  }
+
+  def rorExportCard(event: MovementHistoryEvent)(implicit movement: GetMovementResponse, messages: Messages): Html = {
+    val optContent: Option[Html] = for {
+      reportOfReceipt <- movement.reportOfReceipt
+      if movement.destinationType == DestinationType.Export
+    } yield {
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply app/viewmodels/helpers/events/MovementEventHelper.scala:578]: it's basically a guard. If this returns false, anything after it in the `for` section will be skipped, as will the `yield` stuff, so it'll bail out and return None early. See https://docs.scala-lang.org/scala3/book/control-structures.html#using-a-for-expression-as-the-body-of-a-method

File: app/viewmodels/helpers/events/MovementEventHelper.scala [hunk 6]
```diff
@@ -459,4 +471,235 @@ class MovementEventHelper @Inject()(
...
+      buildOverviewPartial(
+        headingId = None,
+        headingTitle = Some(s"movementHistoryEvent.${event.eventType}.rorExport.h2"),
+        cardTitleMessageKey = None,
+        cardFooterHtml = None,
+        summaryListRows = Seq(
+          // TODO: check if destinationOffice is correct
+          Some(summaryListRowBuilder(s"movementHistoryEvent.${event.eventType}.rorExport.customsOfficeCode", reportOfReceipt.destinationOffice))
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply app/viewmodels/helpers/events/MovementEventHelper.scala:586]: Double checked, my assumption was correct - will remove this TODO

**Review by MJCallahanPage — APPROVED** [review]
LGTM 🟢

**Pull request merged**